src/main/java/com/linagora/james/mailets/json/ClassificationRequestBody.java

```diff
@@ -0,0 +1,112 @@
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               *
+ * GNU Affero General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Affero General Public License    *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.       *
+ *******************************************************************************/
+package com.linagora.james.mailets.json;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.message.MessageBuilder;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.james.util.mime.MessageContentExtractor.MessageContent;
+import org.apache.mailet.Mail;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+public class ClassificationRequestBody {
+
+
+    private static final int NO_LINE_LENGTH_LIMIT_PARSING = -1;
+
+    public static ClassificationRequestBody from(Mail mail, UUID messageId) throws MessagingException, IOException {
+        MimeMessage message = mail.getMessage();
+        
+        return new ClassificationRequestBody(messageId,
+                Emailers.from(message.getFrom()),
+                Recipients.from(message),
+                ImmutableList.of(Optional.ofNullable(message.getSubject()).orElse("")),
+                retrieveTextPart(mail));
+    }
+
+    private static String retrieveTextPart(Mail mail) throws IOException, MessagingException {
+        MessageContent messageContent = new MessageContentExtractor()
+                .extract(toMime4jMessage(mail));
+        return mainTextContent(messageContent).orElse("");
+    }
+
+    private static Message toMime4jMessage(Mail mail) throws IOException, MessagingException {
+        ByteArrayOutputStream rawMessage = new ByteArrayOutputStream();
+        mail.getMessage().writeTo(rawMessage);
+
+        return MessageBuilder
+                .create()
+                .use(MimeConfig.custom().setMaxLineLen(NO_LINE_LENGTH_LIMIT_PARSING).build())
+                .parse(new ByteArrayInputStream(rawMessage.toByteArray()))
+                .build();
+    }
+
+    private static Optional<String> mainTextContent(MessageContent messageContent) {
+        return messageContent.getTextBody()
+            .filter(s -> !Strings.isNullOrEmpty(s))
+            .map(Optional::of)
+            .orElse(messageContent.getHtmlBody());
```

Comment:
No need to extract html. Just take the html as main content, and then text if there is no html (as you did in your fixup).